resources.h

```diff
@@ -73,42 +100,129 @@ enum PaletteName {
 	vortecspace,
 	wind,
 	windowsxp,
-
-	aemil,
-	chasnah,
-	dejil,
-	dzune,
-	ellai,
-	evjar,
-	fruit,
-	gimeljoy,
-	gimelsad,
-	gimelYOO,
-	jaela,
-	jergh,
-	kirii,
-	kraza,
-	llema,
-	lotus,
-	loxxe,
-	meazs,
-	metis,
-	nachi,
-	oom,
-	romal,
-	sillh,
-	vette,
 	yette,
 	zehal,
-	zoog,
 
 	_PALETTE_COUNT,
 };
 
+const static std::map<PaletteGroup, std::vector<PaletteName>> PALETTES_BY_GROUP({
+	{ PaletteGroup::Canon, {
+		PaletteName::aemil,
+		PaletteName::dzune,
+		PaletteName::ellai,
+		PaletteName::evjar,
+		PaletteName::gimeljoy,
+		PaletteName::gimelsad,
+		PaletteName::gimelYOO,
+		PaletteName::jaela,
+		PaletteName::jergh,
+		PaletteName::kirii,
+		PaletteName::kraza,
+		PaletteName::llema,
+		PaletteName::lotus,
+		PaletteName::loxxe,
+		PaletteName::meazs,
+		PaletteName::metis,
+		PaletteName::romal,
+		PaletteName::sillh,
+		PaletteName::vette,
+		PaletteName::zoog,
+	}},
+	{ PaletteGroup::NonCanon, {
+		PaletteName::autumn,
+		PaletteName::ascent,
+		PaletteName::azul,
+		PaletteName::bliss,
+		PaletteName::chasnah,
+		PaletteName::crystal,
+		PaletteName::dejil,
+		PaletteName::follow,
+		PaletteName::friend_,
+		PaletteName::fruit,
+		PaletteName::home,
+		PaletteName::moonflower,
+		PaletteName::nachi,
+		PaletteName::oom,
+		PaletteName::peace,
+		PaletteName::power,
+		PaletteName::purpleflower,
+		PaletteName::radiance,
+		PaletteName::redmoondesert,
+		PaletteName::ripple,
+		PaletteName::stonehenge,
+		PaletteName::tulips,
+		PaletteName::vortecspace,
+		PaletteName::wind,
+		PaletteName::windowsxp,
+		PaletteName::yette,
+		PaletteName::zehal,
+	}},
+	{ PaletteGroup::All, {	// :lkfear:
```

Comment:
Let's do whatever we can to avoid repeating ourselves here since this is just asking for trouble. Here's an idea: since these palette names are just the numbers from 0 to 47 or whatever, let's just generate it at run time with a function:

```c++
inline std::vector<PaletteName> make_all_palettes() {
    std::vector<PaletteName> all_palettes((size_t) PaletteName::_PALETTE_COUNT);
    std::generate(all_palettes.begin(), all_palettes.end(), [n = 0]() mutable { return (PaletteName) n++; });
    return all_palettes;
}

// ...
{ PaletteGroup::All, make_all_palettes() }
```

This is some of the weirdest fucking c++ I've ever seen but it does seem to work. I stole it from [here](https://en.cppreference.com/w/cpp/algorithm/generate).